DOCSP-47961 v2.1 release (#465)
mongodb/docs-golang: source/whats-new.txt

* :ref:`Version 1.12.1 <version-1.12.1>`
* :ref:`Version 1.12 <version-1.12>`

.. _golang-version-2.1:

What's New in 2.1
-----------------

The 2.1 {+driver-short+} release includes the following improvements
and fixes:

> - You can use the new `Client.BulkWrite <{+api+}/mongo#Client.BulkWrite>`__ method 

Reviewer: ```suggestion
- Add the `Client.BulkWrite <{+api+}/mongo#Client.BulkWrite>`__ method 
```

>   to perform many insert, update and delete operations on multiple databases and 

Reviewer: ```suggestion
  to perform many insert, update, and delete operations on multiple databases and 
```
Let's add the serial comma here.

Author: For some reason, the repo is set up to fail a build check if there are Oxford commas :( I agree though; I will add back the comma

Reviewer: Hm, strange decision. Does the build check prevent us from adding the Oxford comma?

  collections in one request.

- The new `bson.Vector <{+api+}/bson#Vector>`__ type makes inserting and querying 
  vector data using :atlas:`Atlas Vector Search </atlas-vector-search/vector-search-overview/>` 
  easier and more efficient.

> - Extends the `ServerError <{+api+}/mongo#ServerError>`__ interface to include 
>   ``ErrorCodes``, which returns a list of deduplicated error codes returned by 
>   the server during the lifetime of operation execution.
> 
> - Adds ths ``sort`` option to `UpdateOneOptions <{+api+}/mongo/options#UpdateOneOptions>`__ 
>   and `ReplaceOptions <{+api+}/mongo/options#ReplaceOptions>`__ for the ``updateOne`` 
>   and ``replaceOne`` operations. 

Reviewer: I'd recommend making the bullet points consistent in their form by doing something like this:
```suggestion
- The `ServerError <{+api+}/mongo#ServerError>`__ interface is extended to include 
  ``ErrorCodes``, which returns a list of deduplicated error codes returned by 
  the server during the lifetime of operation execution.

- The ``sort`` option is added to `UpdateOneOptions <{+api+}/mongo/options#UpdateOneOptions>`__ 
  and `ReplaceOptions <{+api+}/mongo/options#ReplaceOptions>`__ for the ``updateOne`` 
  and ``replaceOne`` operations. 
```
(Also, there was a typo: "ths" instead of "the")

Reviewer: Good to know. The changes you made address the issue. Can we do a similar thing for the first bullet point?

Reviewer: Added a code suggestion on that line. There are other valid ways of doing it, but the main point is to make it consistent with the other three bullet points by leading with a verb in the third person singular.

Author: Accepted your change, thanks!

The 2.1 {+driver-short+} release includes the following improvements
and fixes:

- Adds the new `Client.BulkWrite <{+api+}/mongo#Client.BulkWrite>`__ method 
  to make performing many insert, update, and delete operations on multiple databases and 

>   collections in one request more efficient.

Reviewer: Depending on how you interpret the "more efficient" part it could be slightly inaccurate. How about one of these variants?

> Adds the new `Client.BulkWrite <{+api+}/mongo#Client.BulkWrite>`__ method to provide clients the ability to perform many insert, update, and delete operations on multiple databases and collections in one request.

or

> Introduces the new `Client.BulkWrite <{+api+}/mongo#Client.BulkWrite>`__, enabling clients to perform multiple insert, update, and delete operations across databases and collections in a single request.

Author: I like the second one, thanks! Also added 'Introduce' to second bullet
